### PR TITLE
Disallow copy and move

### LINKED ISSUE #4411: use DISALLOW_COPY_AND_MOVE instead.
## Enhancement
reference:
https://github.com/oceanbase/oceanbase/blob/master/src/observer/ob_cache_size_calculator.h

In CPUAffinityManager.h and many other classes, they disabled copy and move constructor, maybe we can simpilfy the code like:

````
#define DISALLOW_COPY_AND_MOVE(className) \
    DISALLOW_COPY(className);             \
    DISALLOW_MOVE(className)

#define DISALLOW_COPY(className)      \
    className(const className&) = delete; \
    className& operator=(const className&) = delete

#define DISALLOW_MOVE(className) \
    className(className&&) = delete; \
    className& operator=(className&&) = delete

### PR DESCRIPTION
**Pull request by ZHANGWENTAI**
### What problem does this PR solve?

Issue Number: ref #4411

Problem Summary: use DISALLOW_COPY_AND_MOVE instead.

### What is changed and how it works?

add DISALLOW_COPY, DISALLOW_MOVE, DISALLOWCOPY_AND_MOVE macro, and used in the files listed in #4411.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Common/CPUAffinityManager.h [hunk 1]
```diff
@@ -123,9 +126,18 @@ class CPUAffinityManager
...
 public:
-    CPUAffinityManager(const CPUAffinityManager &) = delete;
-    CPUAffinityManager & operator=(const CPUAffinityManager &) = delete;
-    CPUAffinityManager(CPUAffinityManager &&) = delete;
-    CPUAffinityManager & operator=(CPUAffinityManager &&) = delete;
+#define DISALLOW_COPY_AND_MOVE(ClassName) \
+    DISALLOW_COPY(ClassName);             \
+    DISALLOW_MOVE(ClassName)
+
+#define DISALLOW_COPY(ClassName)           \
```
Comment (ywqzzy) [inline dbms/src/Common/CPUAffinityManager.h:133]: You can write the code in /dbms/src/Common/nocopyable.h

Comment (ywqzzy) [reply dbms/src/Common/CPUAffinityManager.h:133]: You can replace it directly. Maybe you can list some of the classes need to refactor in the issue and create some sub-issue, and you refactor 3 or 4 of them in one pr.

Comment (JaySon-Huang) [reply dbms/src/Common/CPUAffinityManager.h:133]: Like @ywqzzy said, the macro `DISALLOW_COPY`/`DISALLOW_MOVE `/`DISALLOW_COPY_AND_MOVE` should be defined in a common place like `dbms/src/Common/noncopyable.h`.

And I wonder whether we can define `DISALLOW_COPY_AND_MOVE` before `DISALLOW_COPY`/`DISALLOW_MOVE `?

**Comment by ZHANGWENTAI** [comment]
I found a mistake in dbms/src/Common/ExecutableTask.h, let me fix it and recommit.

**Comment by ZHANGWENTAI** [comment]
Can anyone review this pr? thx :)

**Review by ywqzzy — APPROVED** [review]
LGTM. Good Job!

**Comment by Lloyd-Pottiger** [comment]
And why do this? :joy:
```cpp
- std::string toString() const { return "Not Support"; }
+ std::string toString() const
+ {
+      return "Not Support";
+  }
```
honestly, I like the previous code style which makes code more compact with high readability. :smile:

**Comment by ywqzzy** [comment]
> And why do this? 😂
> 
> ```c++
> - std::string toString() const { return "Not Support"; }
> + std::string toString() const
> + {
> +      return "Not Support";
> +  }
> ```
> 
> honestly, I like the previous code style which makes code more compact with high readability. 😄


Maybe 

> And why do this? 😂
> 
> ```c++
> - std::string toString() const { return "Not Support"; }
> + std::string toString() const
> + {
> +      return "Not Support";
> +  }
> ```
> 
> honestly, I like the previous code style which makes code more compact with high readability. 😄

Maybe the formatter automatically did the change.

**Comment by Lloyd-Pottiger** [comment]
> Maybe the formatter automatically did the change.

we have enable `AllowShortFunctionsOnASingleLine` in [.clang-format](https://github.com/pingcap/tiflash/blob/master/.clang-format#L11)

**Comment by ZHANGWENTAI** [comment]
> Thanks for your good job! The two copy operations are independent. Declaring copy constructor does not prevent compiler to generate copy assignment and vice versa. And I found that there are some classes which only make copy constructor `= delete` but you also replace it with `DISALLOW_COPY(className)`. In most cases, it is fine. However, could you please take a deep check to make sure it is all fine? And thanks for your great contribution again!

Actually I have noticed this and commented at [https://github.com/pingcap/tiflash/pull/4578#discussion_r845193883](url),
I have no idea how to check this. If a copy assignment constructor used in any other place,  can CI tests find that and output an error? I need a clue. 
Or we could just leave the "one line" copy constructor `= delete` as it used to be ? It's easy to understand with a comment like ` // disallow copy constructor`.

**Comment by ZHANGWENTAI** [comment]
I will format the code again and modify the comment.

**Comment by Lloyd-Pottiger** [comment]
> > Thanks for your good job! The two copy operations are independent. Declaring copy constructor does not prevent compiler to generate copy assignment and vice versa. And I found that there are some classes which only make copy constructor `= delete` but you also replace it with `DISALLOW_COPY(className)`. In most cases, it is fine. However, could you please take a deep check to make sure it is all fine? And thanks for your great contribution again!
> 
> Actually I have noticed this and commented [above](https://github.com/pingcap/tiflash/pull/4578#discussion_r845193883), I have no idea how to check this. If a copy assignment constructor used in any other place, can CI tests find that and output an error? I need a clue. Or we could just leave the "one line" copy constructor `= delete` as it used to be ? It's easy to understand with a comment like ` // disallow copy constructor`.

Normally, if we have define copy assignment `=delete` and do not explicitly define copy operator and vice versa, then we can just replace it with `DISALLOW_COPY(className)`. Since this pr have pass CI build, it seems your work is fine. Thanks.

**Comment by ZHANGWENTAI** [comment]
> > Maybe the formatter automatically did the change.
> 
> we have enable `AllowShortFunctionsOnASingleLine` in [.clang-format](https://github.com/pingcap/tiflash/blob/master/.clang-format#L11)

I format the code in my local repo by `python3 format-diff.py --diff_from `git merge-base master HEAD``, but it still expend short functions. Is I doing it right?

**Comment by Lloyd-Pottiger** [comment]
> > > Maybe the formatter automatically did the change.
> > 
> > 
> > we have enable `AllowShortFunctionsOnASingleLine` in [.clang-format](https://github.com/pingcap/tiflash/blob/master/.clang-format#L11)
> 
> I format the code in my local repo by `python3 format-diff.py --diff_from `git merge-base master HEAD``, but it still expend short functions. Is I doing it right?

Yes. `AllowShortFunctionsOnASingleLine: Inline` in clang-format means allow not force. Therefore, your style is legal for clang-format.

**Comment by ZHANGWENTAI** [comment]
> > > > Maybe the formatter automatically did the change.
> > > 
> > > 
> > > we have enable `AllowShortFunctionsOnASingleLine` in [.clang-format](https://github.com/pingcap/tiflash/blob/master/.clang-format#L11)
> > 
> > I format the code in my local repo by `python3 format-diff.py --diff_from `git merge-base master HEAD``, but it still expend short functions. Is I doing it right?
> 
> Yes. `AllowShortFunctionsOnASingleLine: Inline` in clang-format means allow not force. Therefore, your style is legal for clang-format, so there is nothing changed :joy:

hello! I have remained the format of short function and commit agian

**Comment by Lloyd-Pottiger** [comment]
> > > > > Maybe the formatter automatically did the change.
> > > > 
> > > > 
> > > > we have enable `AllowShortFunctionsOnASingleLine` in [.clang-format](https://github.com/pingcap/tiflash/blob/master/.clang-format#L11)
> > > 
> > > 
> > > I format the code in my local repo by `python3 format-diff.py --diff_from `git merge-base master HEAD``, but it still expend short functions. Is I doing it right?
> > 
> > 
> > Yes. `AllowShortFunctionsOnASingleLine: Inline` in clang-format means allow not force. Therefore, your style is legal for clang-format, so there is nothing changed 😂
> 
> hello! I have remained the format of short function and commit agian

https://github.com/pingcap/tiflash/pull/4578/files#diff-e49f478fa33ca6648a69a7aba47d9485827d89793ef5784c080375ff2fce5cfb seems have not pushed?

**Comment by ZHANGWENTAI** [comment]
> > > > > > Maybe the formatter automatically did the change.
> > > > > 
> > > > > 
> > > > > we have enable `AllowShortFunctionsOnASingleLine` in [.clang-format](https://github.com/pingcap/tiflash/blob/master/.clang-format#L11)
> > > > 
> > > > 
> > > > I format the code in my local repo by `python3 format-diff.py --diff_from `git merge-base master HEAD``, but it still expend short functions. Is I doing it right?
> > > 
> > > 
> > > Yes. `AllowShortFunctionsOnASingleLine: Inline` in clang-format means allow not force. Therefore, your style is legal for clang-format, so there is nothing changed 😂
> > 
> > 
> > hello! I have remained the format of short function and commit agian
> 
> https://github.com/pingcap/tiflash/pull/4578/files#diff-e49f478fa33ca6648a69a7aba47d9485827d89793ef5784c080375ff2fce5cfb seems have not pushed?

Actually I did 😂. The reason why I expand those functions still is to keep code style consistent in the file and some functions are not short enough, expanding them is more easy to read

**Comment by Lloyd-Pottiger** [comment]
> > > > > > > Maybe the formatter automatically did the change.
> > > > > > 
> > > > > > 
> > > > > > we have enable `AllowShortFunctionsOnASingleLine` in [.clang-format](https://github.com/pingcap/tiflash/blob/master/.clang-format#L11)
> > > > > 
> > > > > 
> > > > > I format the code in my local repo by `python3 format-diff.py --diff_from `git merge-base master HEAD``, but it still expend short functions. Is I doing it right?
> > > > 
> > > > 
> > > > Yes. `AllowShortFunctionsOnASingleLine: Inline` in clang-format means allow not force. Therefore, your style is legal for clang-format, so there is nothing changed 😂
> > > 
> > > 
> > > hello! I have remained the format of short function and commit agian
> > 
> > 
> > https://github.com/pingcap/tiflash/pull/4578/files#diff-e49f478fa33ca6648a69a7aba47d9485827d89793ef5784c080375ff2fce5cfb seems have not pushed?
> 
> Actually I did 😂. The reason why I expand those functions still is to keep code style consistent in the file and some functions are not short enough. Expanding them is easier to read

OK，cool

**Review by Lloyd-Pottiger — APPROVED** [review]

**Comment by ZHANGWENTAI** [comment]
PTAL

**Pull request merged**